client/src/container/DetailsPage/functions.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+export const addToFavorite = async ({ courseId, notification }) => {
+  try {
+    const { data } = await axios.post(`/api/v1/favorite/${courseId}`);
+    const message =
+      data.rowCount === 1 ? 'added to favorite' : 'already in favorite';
+    notification.success({ message });
+  } catch ({ response }) {
+    const message = response.data.message || 'failed to add to favorite !';
+    notification.error({ message });
+  }
+};
+
+export const getCourseDetails = async ({
+  courseId,
+  setIsLoading,
+  setErrorMessage,
+  setCourseDetails,
+  setComments,
+}) => {
+  try {
+    const { data } = await axios.get(`/api/v1/courses/${courseId}`);
+    setCourseDetails(data.courseDetails);
+    setComments(data.comments);
+    setIsLoading(false);
+    setErrorMessage('');
+  } catch ({ response }) {
+    const message = response.data.message || 'sorry, something went wrong !';
```

Comment:
fix error handling as the following : 
```suggestion
  } catch (error) {
    const message;
    if (error.response){
    message = response.data.message;
    } else {
    message =  'sorry, something went wrong !';
    }
```